ts-web-component/src/component/my-todo.ts

```diff
@@ -1,149 +1,155 @@
+    assert(this._r);
+    await this._r.mutate.deleteTodo(e.detail);
   };
 
-  filteredTodos = (filter: string) =>
-    this._list.filter((todo) => {
-      if (filter === "all") {
-        return true;
-      }
-      if (filter === "active") {
-        return !todo.completed;
-      }
-      if (filter === "completed") {
-        return todo.completed;
-      }
-      throw new Error("Unknown filter: " + this._filter);
-    });
-
-  handleFilterClick = (e: Event) => {
-    const $target = e.target as Element;
-    this._filter = $target.id;
-    this._filteredList = this.filteredTodos(this._filter);
+  private _handleFilterClick = (e: Event) => {
+    const target = e.target as Element;
+    this._filter = asFilter(target.id);
+    this._filteredList = this._filteredTodos(this._filter);
     this._render();
   };
 
-  constructor() {
-    super();
-    this._filter = "all";
-    this._list = [];
-    this._filteredList = [];
+  private _handleTodoInputKeyUp = async (e: KeyboardEvent) => {
+    assert(this._newTodoInput);
+    const { value } = this._newTodoInput;
+    if (e.key === "Enter" && value) {
+      this._newTodoInput.value = "";
+      await this._createTodo(value);
+    }
+  };
+
+  private _filteredTodos(filter: Filter) {
+    return this._list.filter((todo) => {
+      switch (filter) {
+        case "all":
+          return true;
+        case "active":
+          return !todo.completed;
+        case "completed":
+          return todo.completed;
+      }
+    });
   }
 
   async connectedCallback() {
-    this.r = await setupReplicache();
-    this.r.subscribe(listTodos, {
+    this.appendChild(templateTodo.content.cloneNode(true));
+    this._newTodoInput = this.querySelector(".new-todo");
+    this._listContainer = this.querySelector(".todo-list");
+    this._itemCount = this.querySelector(".item-count");
+    this._filterLinks = this.querySelectorAll(".filters a");
+    assert(this._newTodoInput);
+    this._newTodoInput.addEventListener("keyup", this._handleTodoInputKeyUp);
+
+    for (const filter of this._filterLinks) {
+      filter.addEventListener("click", this._handleFilterClick);
+    }
+
+    this._r = await setupReplicache();
+    this._r.subscribe(listTodos, {
       onData: (data) => {
         this._list = data;
         this._list.sort((a: Todo, b: Todo) => a.sort - b.sort);
-        this._filteredList = this.filteredTodos(this._filter);
+        this._filteredList = this._filteredTodos(this._filter);
         this._render();
       },
     });
-    this.appendChild(templateTodo.content.cloneNode(true));
-    this.newTodoInput = this.querySelector(
-      "#new-todo"
-    ) as HTMLInputElement | null;
-    this.listContainer = this.querySelector("#list-container");
-    this.itemCount = this.querySelector("#itemCount");
-    this.filter = this.querySelectorAll(".filters a");
-    this.newTodoInput!.addEventListener("keyup", (e: KeyboardEvent) => {
```

Comment:
All the other event handlers where "methods". Changed this one to a method as well... but inline arrow functions are fine if we never need to remove the event listener.